src/crimson/os/seastore/cache.cc

```diff
@@ -994,14 +998,15 @@ CachedExtentRef Cache::duplicate_for_write(
 
   if (i->is_exist_clean()) {
     i->version++;
+    i->touched_by = t.get_trans_id();
```

Comment:
I'd like all updates to touched_by to the in the same component.  I suggest either moving this into a CachedExtent method or, better, moving the updates in CachedExtent into Cache and removing the duplicate_for_write/get_mutable_replica, on_initial_write/on_initial_commit, on_delta_write/on_delta_commit distinctions.  There's a similar stray update in FixedKVInternalNode::get_mutable_replica which seems to duplicate the one in CachedExtent::duplicate_for_write.